### PR TITLE
better ordering of search results

### PR DESCRIPTION
**Pull request by kdoh**
Sets the order to dashboards, collections, and then questions. Also fixes some padding and an issue with empty category results.

### CODE REVIEW

**Comment by mazameli** [comment]
Looks great. Only thing I saw was that it's putting pulses at the top of the results. Feels like they should go after collections, but before questions maybe?

File: frontend/src/metabase/home/containers/SearchApp.jsx [hunk 1]
```diff
@@ -48,12 +47,13 @@ export default class SearchApp extends React.Component {
                     .groupBy("model")
                     .pairs()
                     .value()
+                    .reverse() // show dashboards and collections before questions
```
Comment (salsakran) [inline frontend/src/metabase/home/containers/SearchApp.jsx]: `.groupBy("model")` seems to result in an ordering that you're then reversing. Can we just be explicit about the order of the types (as defined by `model`)? 

Feel like this is a little too cute

**Review by salsakran — CHANGES_REQUESTED** [review]

**Review by salsakran — APPROVED** [review]

File: frontend/src/metabase/home/containers/SearchApp.jsx [hunk 2]
```diff
@@ -41,31 +39,85 @@ export default class SearchApp extends React.Component {
...
+                            />
+                          </Link>
+                        ))}
+                      </Card>
+                    </Box>
+                  )}
+                  {types.card && (
+                    <Box mt={2} mb={3}>
+                      <div className="text-uppercase text-grey-4 text-small text-bold my1">
+                        {t`Questions`}
```
Comment (mazameli) [inline frontend/src/metabase/home/containers/SearchApp.jsx]: Looks like there are two `Questions` headings?

**Comment by mazameli** [comment]
lgtm now

**Pull request merged**